### PR TITLE
refactor(metrics): replace kratos metrics with otel metrics

### PR DESCRIPTION
**Pull request by Ccheers**
#### Which issue(s) this PR fixes (resolves / be part of):
#3255

### CODE REVIEW

**Review by tonybase — DISMISSED** [review]

**Review by daemon365 — DISMISSED** [review]

File: middleware/metrics/metrics_test.go [hunk 1]
```diff
@@ -132,25 +191,49 @@ func TestClient(t *testing.T) {
...
 
-	_, err := Client()(nextError)(context.Background(), "test:")
+	// init client handler
+	meter := otel.Meter("test_meter")
+	requests, err := DefaultRequestsCounter(meter, DefaultServerRequestsCounterName)
+	if err != nil {
+		t.Errorf("[DefaultRequestsCounter] something went wrong: %v", err)
+		return
+	}
+	seconds, err := DefaultSecondsHistogram(meter, DefaultServerSecondsHistogramName)
```
Comment (shenqidebaozi) [inline middleware/metrics/metrics_test.go:201]: Does the default need to be exposed? Should be the default implementation when not set by the user

File: middleware/metrics/metrics.go [hunk 2]
```diff
@@ -2,37 +2,94 @@ package metrics
...
+	metricLabelOperation = "operation"
+	metricLabelCode      = "code"
+	metricLabelReason    = "reason"
+)
+
+const (
+	DefaultServerSecondsHistogramName = "server_requests_seconds_bucket"
+	DefaultServerRequestsCounterName  = "server_requests_code_total"
+	DefaultClientSecondsHistogramName = "client_requests_seconds_bucket"
+	DefaultClientRequestsCounterName  = "client_requests_code_total"
```
Comment (shenqidebaozi) [inline middleware/metrics/metrics.go:27]: These don't seem to need to be exposed to the public

File: middleware/metrics/metrics_test.go (hunk 1 above)
Comment (Ccheers) [reply middleware/metrics/metrics_test.go:201]: In the default logic, we cannot assist users in registering the DefaultSecondsHistogramView. Therefore, we set Option default to nil, allowing users to realize the need to register DefaultSecondsHistogramView.

**Review by shenqidebaozi — APPROVED** [review]
LGTM

**Review by hawkingrei — APPROVED** [review]

**Review by shenqidebaozi — APPROVED** [review]
LGTM

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**